runtime/sema/entitlementset_test.go

```diff
@@ -257,17 +270,184 @@ func TestEntitlementSet_Minimize(t *testing.T) {
 	set.Add(e1)
 
 	// NOTE: the set is not minimal
+	assert.False(t, set.minimized)
 	assert.Equal(t, 1, set.Entitlements.Len())
 	assert.Equal(t, 1, set.Disjunctions.Len())
 
 	// Minimize
 
 	set.Minimize()
 
+	assert.True(t, set.minimized)
 	assert.Equal(t, 1, set.Entitlements.Len())
 	assert.Equal(t, 0, set.Disjunctions.Len())
 }
 
+func TestEntitlementSet_MinimallyRepresentable(t *testing.T) {
+	t.Parallel()
+
+	t.Run("no entitlements, no disjunctions", func(t *testing.T) {
+		t.Parallel()
+
+		set := &EntitlementSet{}
+		assert.True(t, set.IsMinimallyRepresentable())
+	})
+
+	t.Run("one entitlement, no disjunctions", func(t *testing.T) {
+		t.Parallel()
+
+		set := &EntitlementSet{}
+
+		e1 := &EntitlementType{
+			Identifier: "E1",
+		}
+		set.Add(e1)
+
+		assert.True(t, set.IsMinimallyRepresentable())
+	})
+
+	t.Run("two entitlements, no disjunctions", func(t *testing.T) {
+		t.Parallel()
+
+		set := &EntitlementSet{}
+
+		e1 := &EntitlementType{
+			Identifier: "E1",
+		}
+		set.Add(e1)
+
+		e2 := &EntitlementType{
+			Identifier: "E2",
+		}
+		set.Add(e2)
+
+		assert.True(t, set.IsMinimallyRepresentable())
+	})
+
+	t.Run("one entitlement, redundant disjunction", func(t *testing.T) {
+		t.Parallel()
+
+		set := &EntitlementSet{}
+
+		e1 := &EntitlementType{
+			Identifier: "E1",
+		}
+
+		e2 := &EntitlementType{
+			Identifier: "E2",
+		}
+
+		set.Add(e1)
+
+		e1e2 := orderedmap.New[EntitlementOrderedSet](2)
+		e1e2.Set(e1, struct{}{})
+		e1e2.Set(e2, struct{}{})
+
+		set.AddDisjunction(e1e2)
+
+		assert.True(t, set.IsMinimallyRepresentable())
+	})
+
+	t.Run("two entitlements, two redundant disjunctions", func(t *testing.T) {
```

Comment:
Can you please also add the visual representation for these combinations as a comment?
e.g: `{(E1, E1) | (E1, E2) | (E1, E2)}` (not sure how correct my example is - also why a comment would be valuable for understanding the test)

Reply:
I think this is `{E1, E3, (E1 | E2), (E3 | E4)}`, minimalized to `{E1, E3}`

Reply:
@SupunS Added in [8133e7f](https://github.com/onflow/cadence/pull/3304/commits/8133e7f0f2842f1474626eef85882efaab1112b0)

Reply:
nice, thank you!